assets/js/theme/core/amp/AmpProductUtils.js

```diff
@@ -58,7 +58,10 @@ export default class ProductUtils {
       this.cartAddAlert.clear();
       this.cartOptionAlert.clear();
 
-      utils.api.productAttributes.optionChange(this.productId, $form.serialize(), (err, response) => {
+      // product template should be passed as a string
+      const productTemplate = template && typeof template === 'string' ? template : null;
+
+      utils.api.productAttributes.optionChange(this.productId, $form.serialize(), productTemplate, (err, response) => {
```

Comment:
**NOTE:** **productTemplate** has been explicitly added as a 3rd argument to `optionChange` method ( see [stencil-utils](https://github.com/bigcommerce/stencil-utils/blob/master/src/api/product-attributes.js#L23) ref). But since there is no specific template can be used in universal way that will suite all themes we use it as an argument here